trl/trainer/grpo_trainer.py

```diff
@@ -1026,18 +1028,34 @@ def _enable_gradient_checkpointing(self, model: PreTrainedModel, args: GRPOConfi
 
     @profiling_decorator
     def _get_last_hidden_state(
-        self, unwrapped_model, input_ids, attention_mask, logits_to_keep, pixel_values, image_grid_thw
+        self,
+        unwrapped_model,
+        input_ids,
+        attention_mask,
+        logits_to_keep,
+        pixel_values=None,
+        image_grid_thw=None,
+        pixel_attention_mask=None,
+        image_sizes=None,
     ):
         if is_peft_model(unwrapped_model):
             unwrapped_model = unwrapped_model.base_model.model
 
         # Build model inputs - check if the model supports logits_to_keep (some models and VLMs don't)
         model_inputs = {"input_ids": input_ids, "attention_mask": attention_mask}
 
+        # For Qwen models:
```

Comment:
Would it make sense to be more explicit by adjusting the `model_inputs` based on the architecture defined in the model config or is that asking for trouble? The reason I ask is because the code below isn't explicitly tested and I wonder if we will be able to easier catch issues with future models by having a dedicated method we unit test

Reply:
For now, I would vote against it. My hope is that VLMs will gradually standardize their inputs over time, which means we’ll be able to handle all (or at least most) of them with a small, generic set of checks like the ones we have here. If we start explicitly branching based on the model architecture in the config, we’ll end up maintaining a growing list of special cases, and that’s exactly the kind of thing that’s going to break as new models appear. But lest see in the future if we want to be explicit here.

> the code below isn't explicitly tested

It's tested though, all cases should be covered: https://github.com/huggingface/trl/blob/979c1885904cd667b5d75211b1eb5a2154effefb/tests/test_grpo_trainer.py#L1601-L1607